### ISSUE TITLE
PIP Install broken: ImportError: No module named lovepacker.bundler

### ISSUE THREAD

**Issue by mhvelplund** [WetDesertRock/lovepacker#1]
I'm running on MacOS 10.14.2 with Python 2.7.15 (default, Oct  2 2018, 11:47:18).
I get the error in the title hwn I try to run lovepacker. Any ideas?

```
$ pip install lovepacker

Collecting lovepacker
  Downloading https://files.pythonhosted.org/packages/53/31/940690d77c798a03afd2392f3d309d16886f668dd63c9ef7f2168b2d3cfc/lovepacker-1.0.4.tar.gz
Collecting zipfile2>=0.0.12 (from lovepacker)
  Downloading https://files.pythonhosted.org/packages/60/ad/d6bc08f235b66c11bbb76df41b973ce93544a907cc0e23c726ea374eee79/zipfile2-0.0.12-py2.py3-none-any.whl (44kB)
    100% |████████████████████████████████| 51kB 1.1MB/s
Building wheels for collected packages: lovepacker
  Running setup.py bdist_wheel for lovepacker ... done
  Stored in directory: /Users/mhv/Library/Caches/pip/wheels/1d/e6/29/4ab0696abadf6488631f24a239b40d9a6779647cba10d5e78a
Successfully built lovepacker
Installing collected packages: zipfile2, lovepacker
Successfully installed lovepacker-1.0.4 zipfile2-0.0.12

$ lovepacker
Traceback (most recent call last):
  File "/usr/local/bin/lovepacker", line 8, in <module>
    import lovepacker.bundler as bundler
ImportError: No module named lovepacker.bundler
```

**Comment by mhvelplund** [comment]
Addendum: the README says that lovepacker is a Python 3.x app, but the first line of `/usr/local/bin/lovepacker` is:

```shell
#!/usr/local/opt/python@2/bin/python2.7
```

Something is not right :)

**Comment by WetDesertRock** [comment]
That might actually provide some clues why the pip install hasn't worked for me in the past.

The pip install failing is a known issue, and I've never been able to figure it out completely.

What I suggest doing right now is running it locally (with python 3.x because that is what I was targeting) instead of installing it. This is what I've had to do in the past.

I'll keep this issue open because I do need to fix the pip install.

**Comment by mhvelplund** [comment]
I got it working by running the install locally and manually installing the zip dependency (note the warning when I run setup: _UserWarning: Unknown distribution option: 'install_requires'_):

```shell
$ python3 setup.py install
/usr/local/Cellar/python/3.6.5/Frameworks/Python.framework/Versions/3.6/lib/python3.6/distutils/dist.py:261: UserWarning: Unknown distribution option: 'install_requires'
  warnings.warn(msg)
running install
running build
running build_py
package init file 'lovepacker/__init__.py' not found (or not a regular file)
creating build
creating build/lib
creating build/lib/lovepacker
copying lovepacker/bundler.py -> build/lib/lovepacker
copying lovepacker/downloadmanager.py -> build/lib/lovepacker
copying lovepacker/macpackager.py -> build/lib/lovepacker
copying lovepacker/winpackager.py -> build/lib/lovepacker
package init file 'lovepacker/__init__.py' not found (or not a regular file)
running build_scripts
creating build/scripts-3.6
copying and adjusting bin/lovepacker -> build/scripts-3.6
changing mode of build/scripts-3.6/lovepacker from 644 to 755
running install_lib
creating /usr/local/lib/python3.6/site-packages/lovepacker
copying build/lib/lovepacker/bundler.py -> /usr/local/lib/python3.6/site-packages/lovepacker
copying build/lib/lovepacker/downloadmanager.py -> /usr/local/lib/python3.6/site-packages/lovepacker
copying build/lib/lovepacker/macpackager.py -> /usr/local/lib/python3.6/site-packages/lovepacker
copying build/lib/lovepacker/winpackager.py -> /usr/local/lib/python3.6/site-packages/lovepacker
byte-compiling /usr/local/lib/python3.6/site-packages/lovepacker/bundler.py to bundler.cpython-36.pyc
byte-compiling /usr/local/lib/python3.6/site-packages/lovepacker/downloadmanager.py to downloadmanager.cpython-36.pyc
byte-compiling /usr/local/lib/python3.6/site-packages/lovepacker/macpackager.py to macpackager.cpython-36.pyc
byte-compiling /usr/local/lib/python3.6/site-packages/lovepacker/winpackager.py to winpackager.cpython-36.pyc
running install_scripts
copying build/scripts-3.6/lovepacker -> /usr/local/bin
changing mode of /usr/local/bin/lovepacker to 755
running install_egg_info
Writing /usr/local/lib/python3.6/site-packages/lovepacker-1.0.4-py3.6.egg-info

$ pip3 install zipfile2
Collecting zipfile2
  Cache entry deserialization failed, entry ignored
  Cache entry deserialization failed, entry ignored
  Downloading https://files.pythonhosted.org/packages/60/ad/d6bc08f235b66c11bbb76df41b973ce93544a907cc0e23c726ea374eee79/zipfile2-0.0.12-py2.py3-none-any.whl (44kB)
    100% |████████████████████████████████| 51kB 1.1MB/s
Installing collected packages: zipfile2
Successfully installed zipfile2-0.0.12
You are using pip version 9.0.3, however version 18.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.

$ /usr/local/bin/lovepacker
usage: lovepacker [-h] [-n GAMENAME] [-o BUILDPATH] [-l LOVEVERSION] [-L]
                  [-i IDENTIFIER] [-c CACHEPATH]
                  [--ignore [IGNOREPATTERNS [IGNOREPATTERNS ...]]]
                  [--processors [PROCESSORS [PROCESSORS ...]]]
                  [--config CONFIG]
                  [--loglevel {CRITICAL,ERROR,WARNING,INFO,DEBUG}]
                  basepath
```

**Comment by mhvelplund** [comment]
... correction I got it working after the steps above, _and_ installing `luajit`:)

**Comment by WetDesertRock** [comment]
I think this means I need to investigate the dependency installation, the hashbang changing, and forcing the installation to use python 3

**Comment by WetDesertRock** [comment]
This should be fixed by a number of recent commits (mainly f7316ceb4bd560faee8dda038dc5901686b4e129 and 09521b2fe1f6852d97e9226aad96fdaec69d591f).

Closing issue. If you cannot verify this fix then we can reopen this (I'd assume)

**Comment by mhvelplund** [comment]
This looks fixed:

```
$ pip3 install -I lovepacker
Collecting lovepacker
  Using cached https://files.pythonhosted.org/packages/a3/30/95f10b81ae7cf450412b56aea94a3fabcf6c5b6855092f7c2b5028d91234/lovepacker-1.0.5.tar.gz
Collecting zipfile2>=0.0.12 (from lovepacker)
  Using cached https://files.pythonhosted.org/packages/60/ad/d6bc08f235b66c11bbb76df41b973ce93544a907cc0e23c726ea374eee79/zipfile2-0.0.12-py2.py3-none-any.whl
Building wheels for collected packages: lovepacker
  Running setup.py bdist_wheel for lovepacker ... done
  Stored in directory: /Users/mhv/Library/Caches/pip/wheels/e4/34/f4/d59257f592c70c9c1583a595bb8c46b55f4f550707ce879d20
Successfully built lovepacker
Installing collected packages: zipfile2, lovepacker
Successfully installed lovepacker-1.0.5 zipfile2-0.0.12
```
